plugin/db/mysql/restore.go

```diff
@@ -533,6 +539,21 @@ func (driver *Driver) downloadBinlogFilesOnServer(ctx context.Context, binlogFil
 
 // FetchAllBinlogFiles downloads all binlog files on server to `binlogDir`.
 func (driver *Driver) FetchAllBinlogFiles(ctx context.Context) error {
+	// Ensure that there's at most one ongoing downloading process for the current MySQL instance.
+	muBinlogFile.Lock()
+	if _, ok := binlogFileTasks[driver.connectionCtx.InstanceName]; ok {
+		log.Debug("There is another binlog file downloading process for this instance, skip downloading", zap.String("instance", driver.connectionCtx.InstanceName))
+		muBinlogFile.Unlock()
+		return nil
+	}
+	binlogFileTasks[driver.connectionCtx.InstanceName] = true
+	muBinlogFile.Unlock()
+	defer func() {
+		muBinlogFile.Lock()
+		delete(binlogFileTasks, driver.connectionCtx.InstanceName)
```

Comment:
We use some goroutines to download binlog files and we don't wait for them (server/backup_runner.go line#85).
And then we delete the key in `binlogFileTasks` here.  When the next cycle expires, will there be two goroutines downloading binlog for the same instance?

Reply:
They'll exit when finding that another binlog file downloading task is running.
Anyway, it's a bad idea to leave the goroutines uncontrolled. I'll set a timeout on them.